src/tests/multihost/ad/test_login_attr.py

```diff
@@ -0,0 +1,336 @@
+    session_multihost.client[0].run_command(
+        f'sed -i  "s|{dom_section}|domain/sssd_domain|g" /etc/sssd/sssd.conf',
+        raiseonerr=False
+    )
+    client.sssd_conf('sssd', {'domains': 'sssd_domain', })
+
+    def remove_sssd_domain():
+        """ Remove sssd_domain"""
+        session_multihost.client[0].run_command(
+            'mv -f /etc/sssd/sssd.conf.sssd_domain /etc/sssd/sssd.conf',
+            raiseonerr=False
+        )
+    request.addfinalizer(remove_sssd_domain)
+
+
+@pytest.mark.tier1_3
+@pytest.mark.adloginattr
+@pytest.mark.usefixtures("joinad")
+class TestADLoginAttributes:
+    """Automated Test Cases for AD Schema ported from bash"""
+
+    @staticmethod
+    def test_0001_login_by_samaccountname(multihost, create_aduser_group):
+        """test_0001_login_by_samaccountname
+
+        :title: IDM-SSSD-TC: ad_provider: ad_login_attr: Default behaviour
+          by login with sAMAccountName
+        :id: ba681368-eefe-459f-906e-1b32d405d496
+        :setup:
+          1. Configure sssd with ldap_user_principal = userPrincipalName
+        :steps:
+          1. Run getent passwd for the user.
+          2. Run su for the user with short name (without domain).
+          3. Run ssh for the user with short name (without domain).
+        :expectedresults:
+          1. User is found.
+          2. Su command succeeds.
+          3. Ssh command succeeds.
+        :customerscenario: False
+        """
+        # Configure sssd
+        multihost.client[0].service_sssd('stop')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        client.backup_sssd_conf()
+        dom_section = f'domain/{client.get_domain_section_name()}'
+        sssd_params = {
+            'ldap_id_mapping': 'True',
+            'use_fully_qualified_names': 'False',
+            'debug_level': '9',
+            'id_provider': 'ad',
+            'ad_domain': multihost.ad[0].domainname.lower(),
+            'ad_server': multihost.ad[0].hostname,
+            'ldap_schema': 'ad',
+            'ldap_user_principal': 'userPrincipalName',
+        }
+        client.sssd_conf(dom_section, sssd_params)
+        client.sssd_conf('sssd', {'services': 'nss, pam, ssh'})
+        client.clear_sssd_cache()
+
+        # Create AD user
+        (aduser, _) = create_aduser_group
+
+        # Search for the user
+        usr_cmd = multihost.client[0].run_command(
+            f'getent passwd {aduser}',
+            raiseonerr=False
+        )
+
+        # Run su command
+        su_result = client.su_success(aduser)
+        # Run ssh command
+        ssh_result = client.auth_from_client(aduser, 'Secret123') == 3
+
+        client.restore_sssd_conf()
+        client.clear_sssd_cache()
+
+        # Evaluate test results
+        assert usr_cmd.returncode == 0, f"User {aduser} was not found."
+        assert su_result, "The su command failed!"
+        assert ssh_result, "The ssh login failed!"
```

Comment:
Asserts do not match with the expectedresults from docsting.

Reply:
The message behind the assert is error message to show in the case of failure so they actually match.